src/toil/fileStores/__init__.py

```diff
@@ -33,32 +34,35 @@ class FileID(str):
     def __new__(cls, fileStoreID, *args):
         return super(FileID, cls).__new__(cls, fileStoreID)
 
-    def __init__(self, fileStoreID, size):
+    def __init__(self, fileStoreID, size, executable=False):
         # Don't pass an argument to parent class's __init__.
         # In Python 3 we can have super(FileID, self) hand us object's __init__ which chokes on any arguments.
         super(FileID, self).__init__()
         self.size = size
+        self.executable = executable
 
     def pack(self):
         """
         Pack the FileID into a string so it can be passed through external code.
         """
-        return '{}:{}'.format(self.size, self)
+        return '{}:{}:{}'.format(self.size, int(self.executable), self)
 
     @classmethod
     def forPath(cls, fileStoreID, filePath):
-        return cls(fileStoreID, os.stat(filePath).st_size)
+        executable = os.stat(filePath).st_mode & stat.S_IXUSR != 0
+        return cls(fileStoreID, os.stat(filePath).st_size, executable)
 
     @classmethod
     def unpack(cls, packedFileStoreID):
         """
         Unpack the result of pack() into a FileID object.
         """
 
-        # Find the separating character (first instance)
-        sepPos = packedFileStoreID.find(':')
+        # Separate each part of fileID
+        vals = packedFileStoreID.split(':')
```

Comment:
Adam's comment on this still needs to be addressed I think: https://github.com/DataBiosphere/toil/pull/3322/files#r534339567